addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRule.java

```diff
@@ -383,12 +403,16 @@ private void raiseAlert(
             int risk,
             String evidence,
             boolean multipleCsp,
+            boolean hasReportUri,
             String policy) {
         String alertName = StringUtils.isEmpty(name) ? getName() : getName() + ": " + name;
-        String otherInfo =
-                multipleCsp
-                        ? Constant.messages.getString(MESSAGE_PREFIX + "otherinfo", policy)
-                        : "";
+        String otherInfo = "";
+        if (multipleCsp) {
+            otherInfo = Constant.messages.getString(MESSAGE_PREFIX + "otherinfo", policy);
+        }
```

Comment:
This could be done as an `else if` of the following, it's clearer that the `hasReportUri` takes precedence that way.